### PR TITLE
[SYCLomatic] Support migration of 5 asm instructions

### PR DESCRIPTION
**Pull request by yihwang-nv**

### CODE REVIEW

File: clang/lib/DPCT/APINames_ASM.inc [hunk 1]
```diff
@@ -58,7 +58,7 @@ ENTRY("copysign", "copysign", true, NO_FLAG, P1, "Successful")
 ENTRY("cos", "cos", true, NO_FLAG, P1, "Successful")
 ENTRY("cp", "cp", false, NO_FLAG, P1, "Comment")
 ENTRY("createpolicy", "createpolicy", false, NO_FLAG, P1, "Comment")
-ENTRY("cvt", "cvt", false, NO_FLAG, P1, "Comment")
+ENTRY("cvt", "cvt", true, NO_FLAG, P1, "Partial")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_ASM.inc:61]: need test

File: clang/lib/DPCT/Asm/AsmParser.cpp [hunk 2]
```diff
@@ -179,15 +183,59 @@ InlineAsmContext::getTypeFromConstraint(StringRef Constraint) {
...
+  // print raw typename without 'struct' prefix
+  PrintingPolicy Policy(LangOptions{});
+  Policy.adjustForCPlusPlus();
+  std::string TypeName = E->getType().getCanonicalType().getAsString(Policy);
+  return llvm::StringSwitch<InlineAsmBuiltinType *>(TypeName)
+      .Case("__half", getBuiltinType(InlineAsmBuiltinType::TK_f16))
+      .Case("__half2", getBuiltinType(InlineAsmBuiltinType::TK_f16x2))
+      .Case("__half_raw", getBuiltinType(InlineAsmBuiltinType::TK_f16))
+      .Case("__half_raw2", getBuiltinType(InlineAsmBuiltinType::TK_f16x2))
+      .Case("__nv_bfloat16", getBuiltinType(InlineAsmBuiltinType::TK_bf16))
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmParser.cpp:235]:
```suggestion
      .Case("__nv_bfloat16", getBuiltinType(InlineAsmBuiltinType::TK_bf16))
      .Case("__nv_bfloat16_raw", getBuiltinType(InlineAsmBuiltinType::TK_bf16))
```

File: clang/lib/DPCT/Asm/AsmParser.cpp [hunk 3]
```diff
@@ -179,15 +183,59 @@ InlineAsmContext::getTypeFromConstraint(StringRef Constraint) {
...
+      .Case("__nv_bfloat162", getBuiltinType(InlineAsmBuiltinType::TK_bf16x2))
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmParser.cpp:236]:
```suggestion
      .Case("__nv_bfloat162", getBuiltinType(InlineAsmBuiltinType::TK_bf16x2))
      .Case("__nv_bfloat162_raw", getBuiltinType(InlineAsmBuiltinType::TK_bf16x2))
```

**Review by tangjj11 — CHANGES_REQUESTED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by tangjj11 — APPROVED** [review]

**Pull request merged**